Add monitor X support for transmission monitors in the ISIS SANS GUI

AntonPiccardoSelg opened the pull request:
~~**Do not merge or test before #20229 has been merged**~~
Fixes #20275

This PR addresses two issues:
1. Dynamic addition of mask files via the GUI
2. Being able to not only specify M3 or M4 but to be able to specify any monitor.

# To Test

#### Test  for "Dynamic addition of mask files via the GUI"

Please find test data here: smb://olympic/babylon5/Scratch/Anton/GUI_TEST
Add the data to the Mantid path

1. Open the ISIS SANS v2 experimental interface
1. Select `SANS2D` as the instrument (below the data table)
2. For the user file load: USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT.txt
3. For the batch file load: batch_file.csv
4. Go the `Settings>Mask` tab
5. Ib the `Mask File Input` line edit field click Browse and browse to `MASK_SANS2D_REAR_Edges_16Mar2015.xml`. Add it
  * Confirm that it appears in the `Masking Information` table at the bottom
6. Go to the `Settings>State Diagnostic` tab and press `Update row selection
7.  Check `mask->mask_files` and confirm that the path to the mask file is at the end of the list


#### Test  for MX monitor for transmission

Please find test data here: smb://olympic/babylon5/Scratch/Anton/GUI_TEST
Add the data to the Mantid path

1. Open the ISIS SANS v2 experimental interface
1. Select `SANS2D` as the instrument (below the data table)
2. For the user file load: USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT.txt
3. For the batch file load: batch_file.csv
4. Go the `Settings>Adjustment` tab
  * Confirm that `Transmission monitor` shows 3
5. Set it to 5
6. Go to the `Settings>State Diagnostic` tab and press `Update row selection
7.  Check `adjustment->calculate_transmission->transmission_monitor` and confirm that it has been set to 5. This does not make sense for SANS2D and will most likely lead to useless reductions, but this feature has been explicitly requested, especially for ZOOM and LARMOR


**Release Notes** 
No release notes required, since the GUI is being release in this cycle

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

NickDraper commented:
I'm not confident to handle this merge for you, it will have to wait for your return.

AnthonyLim23 commented on scripts/test/SANS/gui_logic/run_tab_presenter_test.py:
```diff
@@ -302,6 +302,37 @@ def test_that_populates_the_property_manager_data_service_when_processing_is_cal
+        view, _, _ = create_mock_view(user_file_path, batch_file_path)
+        # We just use the sample_user_file since it exists.
+        view.get_mask_file = mock.MagicMock(return_value=user_file_path)
+
```
I have just had a quick look and the above code seems to be repeated. Is it worth creating moving it to setup or creating a setup function?

AnthonyLim23 commented on scripts/test/SANS/gui_logic/state_gui_model_test.py:
```diff
@@ -299,14 +299,14 @@ def test_that_transmission_monitor_can_be_set(self):
-        state_gui_model.transmission_m4_shift = 234
-        self.assertTrue(state_gui_model.transmission_m4_shift == 234)
+        state_gui_model.transmission_mn_shift = 234
+        self.assertTrue(state_gui_model.transmission_mn_shift == 234)
```
why 234?

AntonPiccardoSelg replied:
I am setting the number on the `state_gui_model` object and ensuring that it has been set here. `234` is a just a number for testing.

AnthonyLim23 replied:
is the number likely to ever change? Im just wondering if it may be worth assigning it to a variable, so if it does change then only one number has to be edited.

AntonPiccardoSelg replied:
The number itself is irrelevant. This is a test, no variables needed.

AnthonyLim23 replied:
ok.

AnthonyLim23 commented:
On the first test, step 6 I cannot add the mask. The add button does not appear to do anything.

AntonPiccardoSelg commented:
@AnthonyLim23 
* Refactored test as suggested
* Disabled the line edit fields for the user and the batch file
* I made use of native file dialogs which allow for network storage access. However this is done in #20295 since I refactored the save dialogs in that PR anyway.
